components/SnippetForm.js

```diff
@@ -40,14 +64,64 @@ function SnippetForm({ onSubmit, formName, defaultData }) {
         placeholder="description of the code"
         defaultValue={defaultData?.description}
       ></textarea>
-      <label htmlFor="link">Link:</label>
-      <input
-        type="text"
-        id="link"
-        name="link"
-        placeholder="type your link here"
-        defaultValue={defaultData?.link}
-      />
+
+      {links.length <= 1 ? (
+        <div>
+          {links.map((linkObject) => (
+            <div key={linkObject.id}>
+              <label htmlFor={linkObject.id}>Link:</label>
+              <input
+                type="text"
+                id={linkObject.id}
+                placeholder="www."
+                defaultValue={linkObject.value}
+                onChange={(event) =>
+                  handleLinkChange(linkObject.id, event.target.value)
+                }
+              />
+            </div>
+          ))}
+
+          <SmallButton
+            type={"button"}
+            onClick={handleAddLink}
+            buttonIcon={"simple-line-icons:plus"}
+            buttonName={"Add another link"}
+            ariaLabel={"hidden"}
+          />
```

Comment:
if you change the divs to ul and li, I would move this out of the list.

This will most likely have an effect on the styling, but is more semantic